ecosystem/typescript/sdk/src/transaction_builder/transaction_builder.test.ts

```diff
@@ -0,0 +1,356 @@
+    );
+  },
+  30 * 1000,
+);
+
+test(
+  "serialize script function payload with type args but no function args",
+  async () => {
+    const moduleName = new AptosTypes.ModuleId(hexToAccountAddress(ADDRESS1), new AptosTypes.Identifier("Coin"));
+
+    const token = new AptosTypes.TypeTagVariantstruct(
+      new AptosTypes.StructTag(
+        hexToAccountAddress(ADDRESS4),
+        new AptosTypes.Identifier("TestCoin"),
+        new AptosTypes.Identifier("TestCoin"),
+        [],
+      ),
+    );
+
+    const scriptFunctionPayload = new AptosTypes.TransactionPayloadVariantScriptFunction(
+      new AptosTypes.ScriptFunction(moduleName, new AptosTypes.Identifier("fake_func"), [token], []),
+    );
+
+    const rawTxn = new AptosTypes.RawTransaction(
+      hexToAccountAddress(ADDRESS3),
+      BigInt(0),
+      scriptFunctionPayload,
+      BigInt(2000),
+      BigInt(0),
+      BigInt(TXN_EXPIRE),
+      new AptosTypes.ChainId(4),
+    );
+
+    const signedTxn = await sign(rawTxn);
+
+    expect(hexBcsSignedTxn(signedTxn)).toBe(
+      "000000000000000000000000000000000000000000000000000000000a550c18000000000000000003000000000000000000000000000000000000000000000000000000000000122204436f696e0966616b655f66756e63010700000000000000000000000000000000000000000000000000000000000000010854657374436f696e0854657374436f696e0000d0070000000000000000000000000000ffffffffffffffff040020b9c6ee1630ef3e711144a648db06bbb2284f7274cfbee53ffcee503cc1a492004031c5c351ff23b0af2b19d1bafc5ab79794a8ef3fc32ec9cdfc48d3d33252e6c629ff1aabadcf5542bd9832695766a02296588fd118484b6b30f786b986e6c602",
+    );
+  },
+  30 * 1000,
+);
+
+test(
+  "serialize script payload with no type args and no function args",
+  async () => {
+    let script = hexToBytes("a11ceb0b030000000105000100000000050601000000000000000600000000000000001a0102");
+
+    const scriptPayload = new AptosTypes.TransactionPayloadVariantScript(new AptosTypes.Script(script, [], []));
+
+    const rawTxn = new AptosTypes.RawTransaction(
+      hexToAccountAddress(ADDRESS3),
+      BigInt(0),
+      scriptPayload,
+      BigInt(2000),
+      BigInt(0),
+      BigInt(TXN_EXPIRE),
+      new AptosTypes.ChainId(4),
+    );
+
+    const signedTxn = await sign(rawTxn);
+
+    expect(hexBcsSignedTxn(signedTxn)).toBe(
+      "000000000000000000000000000000000000000000000000000000000a550c1800000000000000000126a11ceb0b030000000105000100000000050601000000000000000600000000000000001a01020000d0070000000000000000000000000000ffffffffffffffff040020b9c6ee1630ef3e711144a648db06bbb2284f7274cfbee53ffcee503cc1a4920040d698cec385e8696ca2333c75a40c7fb7be053e022d160dd224d36035001d0ea168db48ab7a88900fdbb94fc12ba3ec16d9c16cf85ea97f5370b55a22025e3b0c",
+    );
+  },
+  30 * 1000,
+);
```

Comment:
So I was thinking a bit more about your golden file idea... we should spend more time evaluating the right way for scaling out tests. I think this is good to unblock what you're doing now. Maybe you could also grab the tests from the Python code?

But with respect to the golden files, maybe we output a json blob that contains the private key, raw transaction inputs, provides a signed transaction in hex bcs, and a name  / description? We have a golden file contain an array of those and make sure our sdks can handle that input.

Reply:
Yeah. My next step would be standardizing the JSON blob. We can generate one-off JSON blobs. We don't have to check in the Rust tests code. We just need to store the JSON blobs somewhere.